src/Wit/Gen/Type.hs

```diff
@@ -35,7 +35,7 @@ prettyTypeDef (Variant (normalizeIdentifier -> name) cases) =
     prettyCase (normalizeIdentifier -> n, tys) = pretty n <+> parens (hsep (punctuate comma (map boxType tys)))
     boxType :: Type -> Doc a
     boxType (SrcPosType _ t) = boxType t
-    boxType (User n) | n == name = pretty $ "Box<" ++ n ++ ">"
+    boxType (Defined n) | n == name = pretty $ "Box<" ++ n ++ ">"
```

Comment:
We have two modes when using type is the problem, at here, neutral value is enough. In the plugin context, we will need to expand defined to its real type value to have expanded representation.